### DISCUSSION TITLE
how does mikro orm determines the order in which to persist entities that depend on one another ?

### DISCUSSION THREAD

**Discussion by BrannJoly** [mikro-orm/mikro-orm#4342] [Q&A]
I have an entity A with a ManyToOne() A.b relationship to entity B

If I create a new managed entity B (withentityManager.create), and flush it, then I can create an entity A, and do A.b=B, and everything will work fine.

However, if I first create entity A, and then do A.b = entityManager.create("B",...), flushing the repo will result in an FK error (mikro orm tries to persist A first, and A.b is a FK to an id that's not yet in table B)

Is this by design ? Or am I missing something?
Thanks

**Comment by B4nan** [comment]
Please provide a reproduction for what you are doing, the order is computed based on your entity definition, as well as the values you provide.

**Reply by B4nan** [reply]
To add a bit more context, the order is computed via https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/unit-of-work/CommitOrderCalculator.ts

And then the execution part is here: https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/unit-of-work/UnitOfWork.ts#L843

The issue you are describing sounds like a problem in the first part, which drives what entities go first when it comes to one specific change set type.

**Reply by B4nan** [reply]
I just faced something similar to what you are talking about when `useDefineForClassFields` is enabled (in tsconfig) - which is generally not supported now, as it disables propagation, maybe that's your problem? This default value for `useDefineForClassFields` changes based on the target, newer targets have that enabled automatically.

**Reply by BrannJoly** [reply]
> I just faced something similar to what you are talking about when `useDefineForClassFields` is enabled (in tsconfig) - which is generally not supported now, as it disables propagation, maybe that's your problem? This default value for `useDefineForClassFields` changes based on the target, newer targets have that enabled automatically.

 "target": "es2017" so that's not the issue here. I'll try to provide a reproduction repo. Thank you for your answers!
